### PR TITLE
Fix false negatives for babel decorators

### PR DESCRIPTION
**Pull request by abdonrd**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/4761

> Is there anything in the PR that needs further explanation?

All info here: https://github.com/stylelint/stylelint/issues/4761#issuecomment-836710963

### CODE REVIEW

**Comment by 43081j** [comment]
fwiw the code should let you override it via a babel config, but it doesn't seem to work...

i think this is because of it using an old plugin name. if you update this to be `@babel/plugin-proposal-decorators`, it works fine and lets you override it via babel config. can you change that too in this PR?

**Review by 43081j — APPROVED** [review]

**Comment by abdonrd** [comment]
Friendly ping to previous committers like @hudochenkov, @jeddy3 and @bartlangelaan to feel a little more confident to be able to merge and release this!

(in fact, I don't know if I can release a new version after merge the PR)

File: extract.js
Comment (43081j) [reply extract.js]: really? i did try it locally and it worked fine, though i altered the file in node_modules directly. i had to install the plugin, though. maybe thats what you were missing?

Comment (abdonrd) [reply extract.js]: I also just edit the `node_modules/@stylelint/postcss-css-in-js/extract.js` directly.
I did not install any dependencies. We should?

Comment (43081j) [reply extract.js]: we really should be using what babel recommend and that seems to be the `@babel/plugin-proposal-decorators` plugin, so i personally would take this opportunity to update to that. but yeah that means we should probably require it as a peer dependency, or tell consumers if they want to use decorators they should install it.

one for the stylelint maintainers to decide i think. separately the parser shouldn't be eating up the babel errors either, so people are aware they're missing the dependency

**Comment by melink14** [comment]
Is there a path forward for this? From what I can tell, even if I install `@babel/plugin-proposal-decorators` it won't do anything until this change is in?

**Comment by abdonrd** [comment]
> Is there a path forward for this? From what I can tell, even if I install `@babel/plugin-proposal-decorators` it won't do anything until this change is in?

I'm waiting an answer from the Stylelint Team (@hudochenkov, @jeddy3 or @bartlangelaan).

**Comment by hudochenkov** [comment]
Can we have a test for this change?

**Comment by abdonrd** [comment]
> Can we have a test for this change?

Sorry, I'm not familiar with it. Can you help me with this? Thanks in advance!

**Comment by puku0x** [comment]
@abdonrd I realized my PR #237 is trying to solve the same problem about decorators.

File: extract.js
Comment (puku0x) [inline extract.js:97]: I'm afraid, this doesn't help in my case.

**Comment by hudochenkov** [comment]
Superseded by https://github.com/stylelint/postcss-css-in-js/issues/236 and released in `0.38.0`.

**Pull request closed without merging**
